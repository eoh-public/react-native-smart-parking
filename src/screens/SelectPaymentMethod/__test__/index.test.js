import React from 'react';
import renderer, { act } from 'react-test-renderer';

import { SelectPaymentMethod } from '../index';

describe('test SelectPaymentMethod container', () => {
  let tree;
  const route = {
    params: {
      key: 'SelectPaymentMethod-g9afw7_GHlJN7405kPiXh',
      name: 'SelectPaymentMethod',
      params: {
        body: {
          arrive_at: '2021-02-02T08:00:00.014Z',
          booking_item: [],
          car: 155,
          is_pay_now: true,
          is_save_car: false,
          num_hour_book: 1,
          parking_id: 3,
          payment_card_id: undefined,
          payment_method: '',
          plate_number: '84H-123.45',
        },
        itemProps: {
          address:
            '2 Nguyễn Bỉnh Khiêm, Bến Nghé, Quận 1, Thành phố Hồ Chí Minh, Việt Nam',
          arrive_at: '3:00 PM, 02/02/2021',
          background: '',
          currency: 'đ',
          leave_at: '4:00 PM, 02/02/2021',
          payment_method: [],
          payment_option: 'Thanh toán trước',
          plate_number: '84H-123.45',
          spot_name: '',
          street: 'Thảo cầm viên parking street',
          time_warning: '3:38 PM - 02/02/2021',
          total_hours: 1,
        },
      },
    },
  };

  test('render SelectPaymentMethod container', () => {
    act(() => {
      tree = renderer.create(<SelectPaymentMethod route={route} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
