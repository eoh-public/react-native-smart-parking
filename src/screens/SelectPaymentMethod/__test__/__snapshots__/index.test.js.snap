// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SelectPaymentMethod container render SelectPaymentMethod container 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "paddingBottom": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.1)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.8)",
          "borderRadius": 5,
          "paddingHorizontal": 32,
          "paddingVertical": 24,
        }
      }
      testID="COMMON_LOADING_ANIMATION"
    >
      <ActivityIndicator
        animating={true}
        color="#00979D"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</View>
`;
